apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -313,12 +313,18 @@ def resize(
             self._handle.resize(newshape)
             return (True, "")
 
-    def tiledbsoma_upgrade_shape(self, newshape: Sequence[Union[int, None]]) -> None:
+    def tiledbsoma_upgrade_shape(
+        self, newshape: Sequence[Union[int, None]], check_only: bool = False
+    ) -> StatusAndReason:
         """Allows the array to have a resizeable shape as described in the TileDB-SOMA
         1.15 release notes.  Raises an error if the new shape exceeds maxshape in
         any dimension. Raises an error if the array already has a shape.
         """
-        self._handle.tiledbsoma_upgrade_shape(newshape)
+        if check_only:
+            return self._handle.tiledbsoma_can_upgrade_shape(newshape)
+        else:
+            self._handle.tiledbsoma_upgrade_shape(newshape)
```

Comment:
Nit: Having separate `tiledbsoma_can_upgrade_shape` and `tiledbsoma_update_shape` here too (instead of the boolean parmeter) will make it more clear what is happening in the calling code.

Reply:
@jp-dark I thought about that but I really didn't want to clutter up the user-facing API with too many new methods

* User-facing has a small number, with an optional `check_only`
* Underlying internals (wrappers, `pybind11`, C++) have the separate check-the-thing and do-the-thing variants